src/toxcore/dht/dht_node.rs

```diff
@@ -110,19 +161,43 @@ impl DhtNode {
                     }
                 },
                 None => {
-                    match self.saddr_v4 {
+                    match self.assoc4.saddr {
```

Comment:
```
                    match self.assoc4.saddr {
                        Some(v4) => Some(SocketAddr::V4(v4)),
                        None => None,
                    }
```

I think this code can be reduced into a smaller one. Assume you have `Option<SocketAddrV4>` in `self.assoc4.sadd` and you need to cast it to `Option<SocketAddr>`. You may apply `self.assoc4.saddr.map(Into::into)` which should cast `SocketAddrV4` to `SocketAddr` inside `Option`